### PR TITLE
fix(dashboard): On desks combobox change refresh the page only once

### PR DESCRIPTION
**Pull request by ioanpocol**

### CODE REVIEW

File: client/app/scripts/superdesk-dashboard/module.js [hunk 1]
```diff
@@ -38,26 +33,19 @@ define([
...
-                                    scope.select(scope.userDesks[0], false);
-                            });
-                        } else {
-                            scope.select(_.find(scope.userDesks, {_id: currentDeskId}), false);
-                        }
+                desks.initialize()
+                .then(function() {
+                    desks.fetchCurrentUserDesks().then(function (userDesks) {
+                    	scope.userDesks = userDesks._items;
+                    	if (!desks.activeDeskId && scope.userDesks) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-dashboard/module.js]: `scope.userDesks` will be `true` even for empty list, it would be better to check `length`

File: client/app/scripts/superdesk-dashboard/module.js [hunk 2]
```diff
@@ -38,26 +33,19 @@ define([
...
+                    		scope.select(scope.userDesks[0], false);
+                    	} else {
+                    		if (desks.getCurrentDesk() !== scope.selected) {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-dashboard/module.js]: here it might be more obvious to test for `!= null`, `scope.selected` is not yet defined anyhow right?

**Comment by petrjasek** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:
